src/Managers/Roles/RemoteHarvesterCreepManager.ts

```diff
@@ -1,117 +1,162 @@
 import MemoryApi from "../../Api/Memory.Api";
 import CreepApi from "Api/Creep.Api";
+import MemoryHelper from "Helpers/MemoryHelper";
 
 // Manager for the miner creep role
 export default class RemoteHarvesterCreepManager {
-
     /**
      * run the remote harvester creep
      * @param creep the creep we are running
      */
     public static runCreepRole(creep: Creep): void {
         if (creep.spawning) {
-            return; // don't do anything until spawned
+            return;
         }
 
-        const targetRoom: Room = Game.rooms[creep.memory.targetRoom];
+        const homeRoom = Game.rooms[creep.memory.homeRoom];
+
+        if (creep.room.memory.defcon > 0) {
```

Comment:
This is going to tell the creep he shouldn't flee once his room is defcon 0. this room will be the home room once it flees back to it.

to fix change this to 
`if(Memory.rooms[creep.room.name] && Memory.rooms[creep.room.name].defcon > 0)`